explanation/application-streaming.md

```diff
@@ -1,39 +1,57 @@
-| Provides the initial place where a client and an instance can start a peer to peer discussion. These 'chat rooms' are called sessions and are central to the rest of the stack. | A middleware between the Gateway and Anbox Management Service (AMS). It secures access to AMS, launches instances, relays status information and more. | A messaging queue to allow components to communicate. | A STUN/TURN server needed for WebRTC to work behind NATS and firewalls. |
+### Stream gateway
 
-The following illustration shows how the Streaming Stack components interact with each other when creating a new streaming session:
+The stream gateway provides the initial place where a client and an instance can start a peer to peer discussion. These ‘chat rooms’ are called sessions and are central to the rest of the stack. The gateway manages these sessions in order to connect the user and the instance and while doing so, it selects and makes available the best machines for the user based on the availability and geographical location of the machines.
 
-![Streaming stack sequence|2400x1350](https://assets.ubuntu.com/v1/a14d6ead-application_streaming_stack.png)
+### Stream agent
 
-## Streaming an Application
+The stream agent is the middleware between the stream gateway and the Anbox Management Service (AMS), It is responsible for securing access to AMS, launching instances, relay status information and more.
 
-WebRTC is a peer to peer protocol, but clients must first find each other through a central server. The server does not need to know about the content of the messages between the clients, it only has to forward the messages from one peer to another.
+### AMS
+
+AMS is the management component of Anbox Cloud and is responsible for managing the applications and instances.
+
+### Coturn
+
+A STUN/TURN server needed for WebRTC to work behind NATS and firewalls.
+
+### NATS
 
-The Stream Gateway exists to enable clients find each other through sessions.Sessions contain user data, application information and more, but most importantly, they provide an entry point for both the client and the instance to start the signalling process. Signalling is a process by which both peers exchange messages and establish optimal codecs, network routes and content types.
+A messaging queue to allow components to communicate.
+
+The following illustration shows how the streaming stack components interact with each other when streaming visual output from an instance:
+
+![Streaming stack sequence|2400x1350](https://assets.ubuntu.com/v1/f6b703b0-Cloud-streaming-stack_v4-1-1.png)
+
+## Streaming an application by creating a session
+
+While streaming an application, session creation can happen in two ways as illustrated in the diagram above. Until the 1.22.0 release, to create Anbox Cloud instances that can be streamed, a session needed to be created on the stream gateway first. Once the session is created, the instance creation will be triggered as detailed in the process below:
```

Comment:
The second sentence is a bit hard to read imo. I think it could be easier if it were the other way around, e.g.:
```suggestion
While streaming an application, session creation can happen in two ways as illustrated in the diagram above. Until the 1.22.0 release, a session needed to be created on the stream gateway first to be able to create Anbox Cloud instances that can be streamed. Once the session is created, the instance creation will be triggered as detailed in the process below:
```
I think it is a bit easier to read, but this is probably very subjective. Wdyt?
(also, I'm not sure if we need to add 'be able to')